index.js

```diff
@@ -1160,6 +1160,50 @@ class Encore {
         return this;
     }
 
+    /**
+     * If enabled, images will be optimized with
+     * imagemin.
+     *
+     * ```
+     * Encore.enableImagemin()
+     * ```
+     *
+     * or configure the imagemin options
+     * https://github.com/tcoopman/image-webpack-loader#options
+     *
+     * ```
+     * Encore.enableImagemin({
+     *     mozjpeg: {
+     *         progressive: true,
+     *         quality: 65
+     *     },
+     *     // optipng.enabled: false will disable optipng
+     *     optipng: {
+     *         enabled: false,
+     *     },
+     *     pngquant: {
+     *         quality: [0.65, 0.90],
+     *         speed: 4
+     *     },
+     *     gifsicle: {
+     *         interlaced: false,
+     *     },
+     *     // the webp option will enable WEBP
+     *     webp: {
+     *         quality: 75
+     *     }
+     * });
+     * ```
+     *
+     * @param {object} options
+     * @return {Encore}
+     */
+    enableImagemin(options = {}) {
```

Comment:
That would work but we have some kind of pattern to set loaders' options using a callback, I think it would be better for consistency to use it there too.

It shouldn't require many changes, basically:
* have `enableImagemin` accept a function as its first argument instead of an object
* store this function instead of an object in `lib/WebpackConfig.js`
* use `applyOptionsCallback` in `lib/loaders/images.js` to apply that function on a "default options" object

Reply:
I actually implemented it that way first 😂 
Then I noticed `enablePreactPreset` uses an options object, `enableTypeScriptLoader` uses a callback, `eslintLoaderOptionsOrCallback` supports both types … I wasn't sure about the best or *latest* approach here?

Reply:
Usually it goes like this:
* Encore-specific options are set using objects that are validated directly when the related method is called (that's the case for `enablePreactPreset()`).
* Loaders/plugins options are set using callbacks that are applied on a default options object when building the Webpack config. Sometimes (for instance in `enableTypeScriptLoader`) an object is also allowed but in reality it's just a shorthand that creates a callback. In this case we don't really check what the callback does unless we have a really good reason to do it (for instance if we know for sure that setting an option breaks another feature)

Reply:
Adjusted accordingly in 700b608eb478b7242faf0d624cbbbf16f9bb6fe5